activesupport/lib/active_support/cache.rb

```diff
@@ -718,6 +718,10 @@ def merged_options(call_options)
             expires_at = call_options.delete(:expires_at)
             call_options[:expires_in] = (expires_at - Time.now) if expires_at
 
+            if call_options[:expires_in]&.negative?
+              ActiveSupport::Cache::Store.logger&.warn("Cache expiration is in the past")
```

Comment:
It would make sense to report his has an handled error with `Rails.error.report`

Reply:
Ah, like with the new stuff from #43472? I'll work something up and report back. Thanks for the review!

Reply:
Also we could probably already put this behind a config flag, and make it the new default for 7.1.

Feel free to ping me if you open such PR.

Reply:
Ok I'll work up in 2 PRs, so we'll have both steps done nicely. I'll ping when ready. Thanks!

Reply:
You can open just one it's fine.

If the setting is enabled: raise

if it's not enabled: Rails.error.report.

Reply:
Oh, gotcha. Thx!

Reply:
Sorry for the delay! Follow up here: https://github.com/rails/rails/pull/45892 -- needs feedback, thank you!